### DISCUSSION TITLE
Pure Direct function

### DISCUSSION THREAD

**Discussion by bakovari** [mvdwetering/yamaha_ynca#149] [Ideas]
Hey,
Great job this integration. Could you please add the PURE DIRECT function to work? That would be nice to set on media player card or trigger it by an automation.
Thanks,
Balage

**Comment by mvdwetering** [comment]
You mention the mediaplayer card. How would you expect it to be shown there?
Because I don't think it is possible to just add any setting to the card.

A separate switch entity seems easy to add.

**Reply by bakovari** [reply]
Separate switch entity would be perfect.

**Comment by mvdwetering** [comment]
Pure Direct has been added in v7.0.0

**Reply by bakovari** [reply]
Thank you! It's great also the speed of reqlization of an idea. I have some new ideas will be shared in a separate thread.

**Comment by bakovari** [comment]
New ideas to consider:
- setting of favourite content to play when switched on or NET radio starts playing. (Bookmarks/my_favourites/...
- buttons for favourite (preset) contents (Tuner and NET radio)
- show the content details on media card

**Reply by mvdwetering** [reply]
For what to do when switched on you could build an automation in Home Assistant, that would not be part of an integration.

To choose content that starts playing when selecting an input would be to implement MediaBrowser support in Home Assistant. 
I looked into that a bit in the past, but that seemed not easy to do. The Home Assistant side seems quite more complicated than just basic media_player is (only ha d a quick look at the docs). But the harder part is the protocol to the receiver. It looks like the only way to do see what content is available is to programmitically browse through the menu page by page. This is pretty slow and might interfere with other things that are active on the receiver.

Buttons for presets would probably also be through MediaBrowser support in Home Assistant. So same as above applies (although Tuner just has numbered presets so maybe that is a bit easier to build in the MediaBrowser)

Not sure what you mean with "show the content details on media card". The integration already exposes all available mediainfo as far as I know. For example station names, artist/song/album names. Availability depends on source. If something is missing let me know and I will see what I can do.

It is up to the frontend to show that information. I think that the Home Assistant frontend mediaplayer card only displays a maximum of 2 attributes. If you want more info shown maybe checkout some custom mediaplayer cards or put in a request with Home Assistant to expose more info on the default mediaplayer card.

e.g. netradio shows station name
![image](https://user-images.githubusercontent.com/732514/229298327-0c78036f-1b08-4d93-b08a-eef95fc40f98.png)

or artist + song when playing music from my phone (PC source), but as you can see in developer tools also album is provided (and seems this song is tagged wrongly as that is not the correct album)
![image](https://user-images.githubusercontent.com/732514/229298400-2775b081-2da9-4821-a348-4661ea61387a.png)
![image](https://user-images.githubusercontent.com/732514/229298474-276707d3-a546-4f45-80ad-3a4bd170a3dc.png)

**Reply by bakovari** [reply]
Ok, I understand. I really appreciate the detailed explanation and your great job of integration.
Thank you!
